Base.rte/Activities/BunkerBreach.lua

```diff
@@ -257,630 +204,307 @@ function BunkerBreach:UpdateActivity()
-							then
-								local range = SceneMan:ShortestDistance(actor.Pos, self.CPUBrain.Pos, false).Magnitude
-								table.insert(Intruders, {Act=actor, score=range+math.random(400)})
-								intruder_tally = intruder_tally + 1
+						self.CPUSpawnDelay = self.CPUSpawnDelay * 0.5;--* dist.Magnitude/searchRadius;
+
+						local closestGuard = MovableMan:GetClosestEnemyActor(targetActor.Team, targetActor.Pos, searchRadius - dist.Magnitude, Vector());
+						if closestGuard and math.random() > dist.Magnitude/searchRadius and closestGuard.AIMode == Actor.AIMODE_SENTRY and closestGuard:GetAlarmPoint() ~= Vector() then
+							--Send a nearby alerted guard after the intruder
+							closestGuard.AIMode = Actor.AIMODE_GOTO;
+							closestGuard:SetAIMOWayPoint(targetActor);
+							self.attackPos = nil;
+							--A guard has been sent, the next unit should spawn faster
+							self.CPUSpawnDelay = self.CPUSpawnDelay * 0.8;
+						else
+							self:CreateDrop(self.CPUTeam);
+							self.CPUSpawnDelay = (60000 - self.difficultyRatio * 30000 + moRatio * 15000) * rte.SpawnIntervalScale;
+							if math.random() < 0.5 then
+								--Change target for the next attack
+								self.attackPos = nil;
 							end
 						end
+					else
+						self.attackPos = nil;
 						
-						if intruder_tally > 0 then
-							local intruder = table.remove(Intruders).Act
-						
-							-- Look for nearby defenders, but ignore brain guards
-							for actor in MovableMan.Actors do
-								if actor.Team == self.CPUTeam and actor.Health > 0 and actor.AIMode == Actor.AIMODE_SENTRY and
-									SceneMan:ShortestDistance(actor.Pos, self.CPUBrain.Pos, false).Magnitude > 85 and
-									(actor.ClassName == "AHuman" or actor.ClassName == "ACrab" or actor.ClassName == "Actor")
-								then
-									local range = SceneMan:ShortestDistance(actor.Pos, self.CPUBrain.Pos, false).Magnitude
-									table.insert(Defenders, {Act=actor, score=range + math.random(100)})
-									defender_tally = defender_tally + 1
-								end
-							end
-							
-							-- If we've found a suitable defender then make it intercept the intruder
-							if defender_tally > 0 then
-								table.sort(Defenders, function(A, B) return A.score > B.score end)	-- the nearest intruder last
-								
-								local defender = table.remove(Defenders).Act
-								if defender and intruder then
-									defender:ClearAIWaypoints()
-									defender.AIMode = Actor.AIMODE_GOTO;
-									defender:AddAIMOWaypoint(intruder)
-								end
-							end
+						if moRatio < 1.25 then
+							self:CreateDrop(self.CPUTeam);
+							self.CPUSpawnDelay = (60000 - self.difficultyRatio * 30000 + moRatio * 15000) * rte.SpawnIntervalScale;
+						else
+							self.CPUSpawnDelay = self.CPUSpawnDelay * 0.9;
 						end
-						
-						self.IntruderAlertTimer:Reset();
 					end
 				end
 			end
+		elseif MovableMan:GetTeamMOIDCount(self.CPUTeam) < 5 then
+			MovableMan:KillAllEnemyActors(self.CPUTeam);
```

Comment:
An AHuman with all its limbs is approx. 6 MOIDs, so one damaged enemy is supposed to trigger victory conditions... I know it's not the cleanest, there should really be a property for total enemies on a specific team (or I just don't know there is one)

also, sometimes dropped items and other shit is given the CPU team

Reply:
Ahh, I thought it might be something like that. Very hacky though lol.
Yeah, there's nothing for that that I know of. You could use MovableMan:GetClosestTeamActor and GetNextTeamActor to check if there's only one actor on a team, but then you'd also have to check if they're missing bits and so on. I guess this'd still be a bit more reliable, and if you chuck it behind a timer it won't have any real performance effect.